### ISSUE TITLE
Secure hardening of web pages

### ISSUE THREAD

**Issue by daN4cat** [opensourcepos/opensourcepos#3217]
We should add Subresource Integrity checks and sourcing control in our web pages.

CSP: https://blog.sqreen.com/what-is-a-csp/

SRI: https://cheatsheetseries.owasp.org/cheatsheets/Third_Party_Javascript_Management_Cheat_Sheet.html

**Comment by jekkos** [comment]
Yes CSP should not be that hard to add and adds a good extra layer against XSS.

**Comment by daN4cat** [comment]
To report issues we can use a basic free subscription to report-uri.io.
For 10000 a month reports it's free.

**Comment by daN4cat** [comment]
CSP scan: https://csp-evaluator.withgoogle.com/

**Comment by objecttothis** [comment]
@daN4cat it appears that CI4 will improve CSP implementation (https://codeigniter.com/user_guide/outgoing/response.html#content-security-policy).  Do you feel like configuring CSP in the ci4-upgrade branch?

**Comment by objecttothis** [comment]
I'm going to add this to the 3.5.0 milestone for now. I suspect that some of the issues go away in 3.4.0 due to the CI4 upgrade.
